src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSemanticFactsService.cs

```diff
@@ -401,5 +401,16 @@ private SymbolInfo GetSymbolInfo(SemanticModel semanticModel, SyntaxNode node, S
 
         public bool IsInsideNameOfExpression(SemanticModel semanticModel, SyntaxNode node, CancellationToken cancellationToken)
             => (node as ExpressionSyntax).IsInsideNameOfExpression(semanticModel, cancellationToken);
+
+        public IMethodSymbol GetImplicitObjectCreationConstructor(SemanticModel semanticModel, SyntaxNode node)
+        {
+            if (node.Kind() == Formatting.SyntaxKindEx.ImplicitObjectCreationExpression)
+            {
+                var info = semanticModel.GetSymbolInfo(node);
+                return (IMethodSymbol)info.Symbol;
```

Comment:
is this what we do for normal constructors?  i.e. this is ignoring candidates in the case of errors?  if that is what we do normally, grat.  but if we normally use any/all of the symbols we get back for a normal constructor, then we should follow that here.

Reply:
also, i don't think this method is needed.  it is necessary when we'll need some sort of funky semantic model call to get specific information (like foreach-info), but for FAR, it could just call .GetSymbolInfo(node) since it already knows it's an implici-obje-expr.

Reply:
I had not thought of that. As far as I understand, the logic for FAR on constructors will only consider non-error symbols (uses `GetSymbolInfo(...).Symbol` only, ignoring candidates):

```
        protected static Func<SyntaxNode, SemanticModel, (bool matched, CandidateReason reason)> GetStandardSymbolsNodeMatchFunction(
            ISymbol searchSymbol, Solution solution, CancellationToken cancellationToken)
        {
            return (node, model) =>
            {
                var symbolInfoToMatch = FindReferenceCache.GetSymbolInfo(model, node, cancellationToken);

                var symbolToMatch = symbolInfoToMatch.Symbol;
                var symbolToMatchCompilation = model.Compilation;

                if (SymbolFinder.OriginalSymbolsMatch(searchSymbol, symbolInfoToMatch.Symbol, solution, null, symbolToMatchCompilation, cancellationToken))
                {
                    return (matched: true, CandidateReason.None);
                }
                else if (symbolInfoToMatch.CandidateSymbols.Any(s => SymbolFinder.OriginalSymbolsMatch(searchSymbol, s, solution, null, symbolToMatchCompilation, cancellationToken)))
                {
                    return (matched: true, symbolInfoToMatch.CandidateReason);
                }
                else
                {
                    return (matched: false, CandidateReason.None);
                }
            };
        }

```

---
In reply to: [422305225](https://github.com/dotnet/roslyn/pull/43645#discussion_r422305225) [](ancestors = 422305225)